File: mobile-flutter-poc/lib/models/dhl_pickup_parcel_request.dart
/// Schlankes Flutter-Gegenstueck zu `DhlPickupParcelRequest.java`
/// (storebackend.dto) - Request fuer `POST
/// /api/stores/{storeId}/dhl/parcels/pickup` (siehe
/// `DhlController.pickupParcel`).
///
/// Bewusst nur das eine Feld, das der Endpoint tatsaechlich erwartet - das
/// Backend validiert den Tracking-Code hierbei selbst NOCHMALS autoritativ
/// gegen die DHL Tracking API (siehe `DhlController.pickupParcel`,
/// "AUTHORITATIVE DHL VALIDATION"), bevor der Status auf `PICKED_UP`
/// gesetzt wird.
class DhlPickupParcelRequest {
  const DhlPickupParcelRequest({required this.trackingCode});

  final String trackingCode;

  Map<String, dynamic> toJson() => {'trackingCode': trackingCode};
}

File: src/main/java/storebackend/dto/DhlPickupParcelRequest.java
package storebackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Request: Paket abholen
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DhlPickupParcelRequest {
    /**
     * Roher Tracking-Code (wird serverseitig normalisiert)
     */
    private String trackingCode;
}
